airbyte-server/src/test/java/io/airbyte/server/handlers/SchedulerHandlerTest.java

```diff
@@ -373,7 +374,7 @@ void testCheckDestinationConnectionFromUpdate() throws IOException, JsonValidati
   @Test
   void testDiscoverSchemaForSourceFromSourceId() throws IOException, JsonValidationException, ConfigNotFoundException {
```

Comment:
I think it would be a good idea to add some tests here that test the cache busting/non-busting cases introduced in this PR

Reply:
I added some tests to cover the cache where:
- the catalog was not in cache
- the catalog is in cache and the cache was not disabled
- the catalog is in cache and the cache was disabled
Please take another look